presto-main/src/main/java/com/facebook/presto/dispatcher/DispatchManager.java

```diff
@@ -268,16 +270,17 @@ private <C> void createQueryInternal(QueryId queryId, String slug, int retryCoun
             }
 
             // decode session
-            session = sessionSupplier.createSession(queryId, sessionContext, warningCollectorFactory);
+            sessionBuilder = sessionSupplier.createSessionBuilder(queryId, sessionContext, warningCollectorFactory);
 
             // prepare query
-            AnalyzerOptions analyzerOptions = createAnalyzerOptions(session, session.getWarningCollector());
-            QueryPreparerProvider queryPreparerProvider = queryPreparerProviderManager.getQueryPreparerProvider(getAnalyzerType(session));
-            preparedQuery = queryPreparerProvider.getQueryPreparer().prepareQuery(analyzerOptions, query, session.getPreparedStatements(), session.getWarningCollector());
+            AnalyzerOptions analyzerOptions = createAnalyzerOptions(sessionBuilder.build(), sessionBuilder.getWarningCollector());
```

Comment:
We're creating new Session instances here and  on the next line.  I think it would be better to use a local variable:
Session tempSession = sessionBuilder.build() 

and pass to both.